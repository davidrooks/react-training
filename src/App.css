.Navigator{  
  position: absolute; 
  width: 100%; 
}

.ImageCarousel{  
  position: relative;
}

#previousImage {
  -webkit-transform: scaleX(-1);
  transform: scaleX(-1);
}



