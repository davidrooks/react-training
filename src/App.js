import React, { Component } from 'react';
import './App.css';
import arrow from './assets/arrow.svg';

class App extends Component {
  constructor(props){
    super(props);
    this.state = {imageIndex: 0, images: [], hasErrored: false};
  }; 

  nextImage = () => {
    if (this.state.imageIndex < this.state.images.length-1) {
      this.setState({imageIndex: this.state.imageIndex + 1});
    } else {
      this.setState({imageIndex: 0});
    }
  }

  previousImage = () => {
    if (this.state.imageIndex === 0) {
      this.setState({imageIndex: this.state.images.length - 1});
    } else {
      this.setState({imageIndex: this.state.imageIndex - 1});
    }
  }

  componentDidMount() {
    fetch('https://pixabay.com/api/?key=9656065-a4094594c34f9ac14c7fc4c39')
    .then(results => {return results.json()})
    .then(data => {       
      if ('hits' in data && Array.isArray(data.hits)) {  
        // TODO parse each image to check for required fields
        this.setState({images: data.hits});
      } else {
        this.setState({hasErrored: true});
      }
    })
    .catch(error => this.setState({hasErrored: true}));
  }

  render() {
    if (this.state.hasErrored){
      return (
        <div className="App">
          <p>Sorry, there was an error, please try again later.</p>
        </div>
      );
    } else if (this.state.images.length > 0)      
      return (                    
        <div className="ImageCarousel">
          <img src={this.state.images[this.state.imageIndex].webformatURL} alt={this.state.images[this.state.imageIndex].tags} />
          <div className="Navigator">   
            <button onClick={this.previousImage} >
              <img id="previousImage" src={arrow} alt='left arrow'/>
            </button>

            <button onClick={this.nextImage}>
              <img id="nextImage" src={arrow} alt='right arrow'  />
            </button>                     
          </div>
        </div>                  
      );
    else {
      return (
        <div className="App">
          <p>Loading...</p>
        </div>
      );
    }
  }
}

export default App;
