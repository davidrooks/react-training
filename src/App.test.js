import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

const images = [
  {webformatURL: 'image1', tags: 'tags1'},
  {webformatURL: 'image2', tags: 'tags2'},
  {webformatURL: 'image3', tags: 'tags3'},
  {webformatURL: 'image4', tags: 'tags4'},
  {webformatURL: 'image5', tags: 'tags5'},
  {webformatURL: 'image6', tags: 'tags6'},
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render loading message when no images and no error', () => {
  const app = shallow(<App />);
  app.setState({hasErrored: false, images:[]});
  expect(app).toMatchSnapshot();
})

it('should render error message if error state', () => {
  const app = shallow(<App />);
  app.setState({hasErrored: true});
  expect(app).toMatchSnapshot();
})

it('should render error message if error state and images have been loaded', () => {
  const app = shallow(<App />);
  app.setState({hasErrored: true, images: [1,2,3,5]});
  expect(app).toMatchSnapshot();
})

it('should display frist image when images are loaded', () =>{
  const app = shallow(<App />);
  app.setState({images});
  expect(app).toMatchSnapshot();
})

it('should display next image when next button clicked', () =>{
  const app = shallow(<App />);
  app.setState({images});
  expect(app.find('button').length).toBe(2);
  app.find('button').last().simulate('click');
  expect(app).toMatchSnapshot();
})

it('should display previous image when previous button clicked', () =>{
  const app = shallow(<App />);
  app.setState({images});
  expect(app.find('button').length).toBe(2);
  app.find('button').first().simulate('click');
  expect(app).toMatchSnapshot();
})