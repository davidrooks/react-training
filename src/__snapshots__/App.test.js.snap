// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display frist image when images are loaded 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="tags1"
          src="image1"
        />,
        <div
          className="Navigator"
        >
          <button
            onClick={[Function]}
          >
            <img
              alt="left arrow"
              id="previousImage"
              src="arrow.svg"
            />
          </button>
          <button
            onClick={[Function]}
          >
            <img
              alt="right arrow"
              id="nextImage"
              src="arrow.svg"
            />
          </button>
        </div>,
      ],
      "className": "ImageCarousel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "tags1",
          "src": "image1",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />
            </button>,
            <button
              onClick={[Function]}
            >
              <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />
            </button>,
          ],
          "className": "Navigator",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "left arrow",
                "id": "previousImage",
                "src": "arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "right arrow",
                "id": "nextImage",
                "src": "arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="tags1"
            src="image1"
          />,
          <div
            className="Navigator"
          >
            <button
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />
            </button>
          </div>,
        ],
        "className": "ImageCarousel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "tags1",
            "src": "image1",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
              >
                <img
                  alt="left arrow"
                  id="previousImage"
                  src="arrow.svg"
                />
              </button>,
              <button
                onClick={[Function]}
              >
                <img
                  alt="right arrow"
                  id="nextImage"
                  src="arrow.svg"
                />
              </button>,
            ],
            "className": "Navigator",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="left arrow"
                  id="previousImage"
                  src="arrow.svg"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "left arrow",
                  "id": "previousImage",
                  "src": "arrow.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="right arrow"
                  id="nextImage"
                  src="arrow.svg"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "right arrow",
                  "id": "nextImage",
                  "src": "arrow.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should display next image when next button clicked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="tags2"
          src="image2"
        />,
        <div
          className="Navigator"
        >
          <button
            onClick={[Function]}
          >
            <img
              alt="left arrow"
              id="previousImage"
              src="arrow.svg"
            />
          </button>
          <button
            onClick={[Function]}
          >
            <img
              alt="right arrow"
              id="nextImage"
              src="arrow.svg"
            />
          </button>
        </div>,
      ],
      "className": "ImageCarousel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "tags2",
          "src": "image2",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />
            </button>,
            <button
              onClick={[Function]}
            >
              <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />
            </button>,
          ],
          "className": "Navigator",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "left arrow",
                "id": "previousImage",
                "src": "arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "right arrow",
                "id": "nextImage",
                "src": "arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="tags2"
            src="image2"
          />,
          <div
            className="Navigator"
          >
            <button
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />
            </button>
          </div>,
        ],
        "className": "ImageCarousel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "tags2",
            "src": "image2",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
              >
                <img
                  alt="left arrow"
                  id="previousImage"
                  src="arrow.svg"
                />
              </button>,
              <button
                onClick={[Function]}
              >
                <img
                  alt="right arrow"
                  id="nextImage"
                  src="arrow.svg"
                />
              </button>,
            ],
            "className": "Navigator",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="left arrow"
                  id="previousImage"
                  src="arrow.svg"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "left arrow",
                  "id": "previousImage",
                  "src": "arrow.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="right arrow"
                  id="nextImage"
                  src="arrow.svg"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "right arrow",
                  "id": "nextImage",
                  "src": "arrow.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should display previous image when previous button clicked 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="tags6"
          src="image6"
        />,
        <div
          className="Navigator"
        >
          <button
            onClick={[Function]}
          >
            <img
              alt="left arrow"
              id="previousImage"
              src="arrow.svg"
            />
          </button>
          <button
            onClick={[Function]}
          >
            <img
              alt="right arrow"
              id="nextImage"
              src="arrow.svg"
            />
          </button>
        </div>,
      ],
      "className": "ImageCarousel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "tags6",
          "src": "image6",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />
            </button>,
            <button
              onClick={[Function]}
            >
              <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />
            </button>,
          ],
          "className": "Navigator",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "left arrow",
                "id": "previousImage",
                "src": "arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "right arrow",
                "id": "nextImage",
                "src": "arrow.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="tags6"
            src="image6"
          />,
          <div
            className="Navigator"
          >
            <button
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                id="previousImage"
                src="arrow.svg"
              />
            </button>
            <button
              onClick={[Function]}
            >
              <img
                alt="right arrow"
                id="nextImage"
                src="arrow.svg"
              />
            </button>
          </div>,
        ],
        "className": "ImageCarousel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "tags6",
            "src": "image6",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
              >
                <img
                  alt="left arrow"
                  id="previousImage"
                  src="arrow.svg"
                />
              </button>,
              <button
                onClick={[Function]}
              >
                <img
                  alt="right arrow"
                  id="nextImage"
                  src="arrow.svg"
                />
              </button>,
            ],
            "className": "Navigator",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="left arrow"
                  id="previousImage"
                  src="arrow.svg"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "left arrow",
                  "id": "previousImage",
                  "src": "arrow.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="right arrow"
                  id="nextImage"
                  src="arrow.svg"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "right arrow",
                  "id": "nextImage",
                  "src": "arrow.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should render error message if error state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Sorry, there was an error, please try again later.
      </p>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Sorry, there was an error, please try again later.",
      },
      "ref": null,
      "rendered": "Sorry, there was an error, please try again later.",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Sorry, there was an error, please try again later.
        </p>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry, there was an error, please try again later.",
        },
        "ref": null,
        "rendered": "Sorry, there was an error, please try again later.",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should render error message if error state and images have been loaded 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Sorry, there was an error, please try again later.
      </p>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Sorry, there was an error, please try again later.",
      },
      "ref": null,
      "rendered": "Sorry, there was an error, please try again later.",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Sorry, there was an error, please try again later.
        </p>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry, there was an error, please try again later.",
        },
        "ref": null,
        "rendered": "Sorry, there was an error, please try again later.",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should render loading message when no images and no error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Loading...
      </p>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Loading...
        </p>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
